src/Compilers/Core/Portable/CommandLine/AnalyzerConfigOptionsResult.cs

```diff
@@ -26,6 +26,24 @@ namespace Microsoft.CodeAnalysis
         /// </summary>
         public AnalyzerOptions AnalyzerOptions { get; }
 
+        /// <summary>
+        /// Options in the form of <c>build_property.[name]</c> can be accessed here by using a key of <c>[name]</c>
+        /// </summary>
+        /// <remarks>
+        /// These are typically added automatically by the build system 
+        /// </remarks>
+        public AnalyzerOptions BuildProperties { get; }
+
+        /// <summary>
+        /// Options in the form of <c>build_metadata.[name]</c> can be accessed here by using a key of <c>[name]</c>
+        /// </summary>
+        /// <remarks>
+        /// These are typically added automatically by the build system, often in the form <c>build_metadata.[itemtype].[metadataname]</c>
+        /// The parsing does not distinguish the <c>[itemtype]</c> or <c>[metadataname]</c>, and treats everything after <c>build_metadata.</c> as an opaque key
+        /// For instance <c>build_metadata.compile.fullpath = value</c> would be accessed here with a key of <c>compile.fullpath</c>
+        /// </remarks>
+        public AnalyzerOptions BuildMetadata { get; }
```

Comment:
Not sure if we want to fully parse these into a two-level dictionary, or keep the `itemtype.metadataname` as an opaque compound key.